classes/submission/PKPSubmissionMetadataFormImplementation.inc.php

```diff
@@ -147,6 +152,25 @@ function execute($submission, $request) {
 		$submission->setRights($this->_parentForm->getData('rights'), null); // Localized
 		$submission->setSource($this->_parentForm->getData('source'), null); // Localized
 		$submission->setCitations($this->_parentForm->getData('citations'));
+		
+		if ($request->getUserVar('submissionRevision')) {
+			$revision = $request->getUserVar('submissionRevision');
+			
+			$revisionSubmissionData = $submissionDao->getById($submission->getId(), null, false, $revision);
+			// get field names of the publication tab (= additionalFieldNames) and update them
+			$publicationTabFieldNames = $submissionDao->getAdditionalFieldNames();
+			
+			foreach($publicationTabFieldNames as $publicationTabFieldName) {
+				$submission->setData($publicationTabFieldName, $revisionSubmissionData->getData($publicationTabFieldName));
+			}
+		} else {
+			$revision = $submissionDao->getLatestRevisionId($submission->getId());
+		}
+
+		if ($request->getUserVar('saveAsRevision') && ($request->getUserVar('saveAsRevision') == 1)) {
```

Comment:
Wouldn't just...

    if ($request->getUserVar('saveAsRevision') ...

...be sufficient? Much more readable IMO.